### ISSUE TITLE
More explicit examples of the symbolic graph

### ISSUE THREAD

**Issue by botev** [dmlc/nnvm#157]
Could you guys provide more examples of how to use the symbolic graph in NVVM. 
At this stage, the examples in your docs do not show for instance how to make/define the update procedures to the parameters from optimizers, how to create models with more than one objective which alternate, taking gradients with respect to different objectives, how to make loops and any control flow. What things it can not do yet?

Any of these are very important for people coming from Theano/Tensorflow to be able to pick up and use it.

**Comment by tqchen** [comment]
We decided to keep the graphir minimum without making nnvm another framework. The initial focus of the compiler project is on fast inference deployment on more devices. 

But NNVM does support training, and we can find some examples in MXNet and [TinyFlow](https://github.com/tqchen/tinyflow)
